lib/eventFactory.js

```diff
@@ -236,15 +234,36 @@ function createBuilds(config) {
         return null;
     }
 
-    // enable pipeline parameters feature
-    if (pipeline.parameters && pipelineConfig.meta && pipelineConfig.meta.parameters) {
-        const allowedPipelineParameters = _.keys(pipeline.parameters);
-        const currentPipelineParameters = _.assign({},
-            pipeline.parameters,
-            pipelineConfig.meta.parameters);
+    // enable pipeline parameters feature, if parameters exist in screwdriver.yaml
+    if (pipeline.parameters) {
+        const allowedParameters = Object.create(null);
+
+        Object.entries(pipeline.parameters).forEach(([name, val]) => {
+            let defaultParameterValue = val;
+
+            if (typeof val === 'object') {
+                defaultParameterValue = val.value;
+            }
+
+            allowedParameters[name] = { value: defaultParameterValue };
```

Comment:
why do we need to make this an object with keyt`value` ?
why can't this be simply ` allowedParameters[name] = defaultParameterValue;`

Reply:
Unified behavior for end users when they access parameters. Users can only access `parameter.user.value`, while `parameter.user` directly is not permitted.

Reply:
what if a user wants both `value` and `description`?

Reply:
we should have a limit on description. But to be consistent with yaml data, let's keep this proposal.